python/pyarrow/_s3fs.pyx

```diff
@@ -105,9 +105,14 @@ cdef class S3FileSystem(FileSystem):
                 raise ValueError(
                     'Cannot pass anonymous=True together with access_key '
                     'and secret_key.')
+
+            if session_token is None:
+                session_token = ""
+
             options = CS3Options.FromAccessKey(
                 tobytes(access_key),
-                tobytes(secret_key)
+                tobytes(secret_key),
+                tobytes(session_token)
```

Comment:
If `session_token` is only used when either `access_key` or `secret_key` is given, then an error should be raised when `session_token and access_key is None and secret_key is None`.

Reply:
Done.